src/reducers/projects.js

```diff
@@ -83,6 +83,21 @@ export function reduceRoot(stateIn, action) {
             projects.get(currentProjectKey),
           );
         }
+      case 'FOCUS_LINE':
+        {
+          const currentProjectKey =
+            stateIn.getIn(['currentProject', 'projectKey']);
+
+          if (isNil(currentProjectKey)) {
+            return stateIn;
+          }
+
+          return projects.updateIn(
+            [currentProjectKey, 'hiddenUIComponents'],
+            hiddenUIComponents =>
+              hiddenUIComponents.remove(`editor.${action.payload.language}`),
+          );
+        }
```

Comment:
@inlinestyle I actually think changing the action payload structure is a pretty strong argument against this approach—from the perspective of the action, you’ve just asked to focus a particular line of code. It shouldn’t need to know anything about “component names” or even that one of the responses to that action is to maximize the editor component in question.